snow/engine/snowman/transitive_test.go

```diff
@@ -1045,47 +1065,115 @@ func TestEngineBlockingChitResponse(t *testing.T) {
+			RequestID: requestID,
+		}
+		require.Equal(missingBlk.ID(), blkID)
+	}
+
+	// Issuing [blockingBlk] will register an issuer job for [blockingBlk]
+	// awaiting on [missingBlk]. It will also send a request for [missingBlk].
+	require.NoError(te.Put(
 		context.Background(),
-		te.Ctx.NodeID,
-		blockingBlk,
-		false,
-		te.metrics.issued.WithLabelValues(unknownSource),
+		peerID,
+		0,
+		blockingBlk.Bytes(),
 	))
 
-	queryRequestID := new(uint32)
-	sender.SendPullQueryF = func(_ context.Context, inVdrs set.Set[ids.NodeID], requestID uint32, blkID ids.ID, requestedHeight uint64) {
-		*queryRequestID = requestID
-		vdrSet := set.Of(vdr)
-		require.Equal(vdrSet, inVdrs)
+	var queryRequest *common.Request
+	sender.SendPullQueryF = func(_ context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, blkID ids.ID, requestedHeight uint64) {
+		require.Nil(queryRequest)
+		require.Equal(set.Of(peerID), nodeIDs)
+		queryRequest = &common.Request{
+			NodeID:    peerID,
+			RequestID: requestID,
+		}
 		require.Equal(issuedBlk.ID(), blkID)
 		require.Equal(uint64(1), requestedHeight)
 	}
 
-	require.NoError(te.issue(
+	// Issuing [issuedBlk] will immediately adds [issuedBlk] to consensus, sets
+	// it as the preferred block, and sends a query for [issuedBlk].
+	require.NoError(te.Put(
 		context.Background(),
-		te.Ctx.NodeID,
-		issuedBlk,
-		false,
-		te.metrics.issued.WithLabelValues(unknownSource),
+		peerID,
+		0,
+		issuedBlk.Bytes(),
 	))
 
-	sender.SendPushQueryF = nil
-	sender.CantSendPushQuery = false
-
-	require.NoError(te.Chits(context.Background(), vdr, *queryRequestID, blockingBlk.ID(), issuedBlk.ID(), blockingBlk.ID()))
+	sender.SendPullQueryF = nil
 
+	// In response to the query for [issuedBlk], the peer is responding with,
+	// the currently pending issuance, [blockingBlk]. The direct conflict of
+	// [issuedBlk] is [missingBlk]. This registers a voter job dependent on
+	// [blockingBlk] and [missingBlk].
+	require.NoError(te.Chits(
+		context.Background(),
+		queryRequest.NodeID,
+		queryRequest.RequestID,
+		blockingBlk.ID(),
+		missingBlk.ID(),
+		blockingBlk.ID(),
+	))
 	require.Len(te.blocked, 2)
-	sender.CantSendPullQuery = false
 
-	require.NoError(te.issue(
+	queryRequest = nil
+	sender.SendPullQueryF = func(_ context.Context, nodeIDs set.Set[ids.NodeID], requestID uint32, blkID ids.ID, requestedHeight uint64) {
+		require.Nil(queryRequest)
+		require.Equal(set.Of(peerID), nodeIDs)
+		queryRequest = &common.Request{
+			NodeID:    peerID,
+			RequestID: requestID,
+		}
+		require.Equal(blockingBlk.ID(), blkID)
+		require.Equal(uint64(1), requestedHeight)
+	}
+
+	vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		switch blkID {
+		case GenesisID:
+			return Genesis, nil
+		case issuedBlk.ID():
+			return issuedBlk, nil
+		case missingBlk.ID():
+			return missingBlk, nil
+		case blockingBlk.ID():
+			return blockingBlk, nil
+		default:
+			return nil, errUnknownBlock
+		}
+	}
```

Comment:
This test was previously pretty broken because the engine assumes that blocks that have been `Verify`ed are `GetBlock`able. So once we `issue`ed `missingBlk`, the engine was in a weird state.